mmedit/models/base_archs/wrapper.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any, List, Optional, Union
+from warnings import warn
+
+from mmengine.model import BaseModule
+
+
+class DiffuserWrapper(BaseModule):
+    """Wrapper for models from HuggingFace Diffusers. This wrapper will be set
+    a attribute called `_module_cls` by wrapping function and will be used to
+    initialize the model structure.
+
+    Example:
+    >>> 1. Load pretrained model from HuggingFace Space.
+    >>> config = dict(
+    >>>     type='ControlNetModel',  # has been registered in `MODELS`
+    >>>     repo_id='lllyasviel/sd-controlnet-canny')
+    >>> controlnet = MODELS.build(config)
+
+    >>> 2. Initialize model structure but do not load pretrained weight
+    >>> config = dict(
+    >>>     type='ControlNetModel',  # has been registered in `MODELS`
+    >>>     repo_id='lllyasviel/sd-controlnet-canny',
+    >>>     no_loading=True)
+    >>> controlnet = MODELS.build(config)
+
+    >>> 3. Loading pretrained model with specific settings (e.g., fp16).
+    >>> config = dict(
+    >>>     type='ControlNetModel',  # has been registered in `MODELS`
+    >>>     repo_id='lllyasviel/sd-controlnet-canny',
+    >>>     init_cfg=dict(type='Pretrained', torch_dtype=torch.float16))
+    >>> controlnet = MODELS.build(config)
+
+    >>> 4. Initialize model with own defined arguments
+    >>> config = dict(
+    >>>     type='ControlNetModel',
+    >>>     in_channels=3,
+    >>>     down_block_types=['DownBlock2D'],
+    >>>     block_out_channels=(20, ),
+    >>>     conditioning_embedding_out_channels=(16, ))
+    >>> controlnet = MODELS.build(config)
+
+    Args:
+        repo_id (Union[str, os.PathLike], optional): The *model id* or path to
+            pretrained model. Please refers to `diffusers.model.modeling_utils.ModelMixin.from_pretrained`.  # noqa
+            If passed, the structure of the model will be initialized as model
+            structure of `repo_id`. Defaults to None.
+        no_loading (bool): Whether loading the pretrained weights of `repo_id`.
+            If True, weight of `repo_id` will not be loaded. Defaults to False.
+        init_cfg (dict or List[dict], optional): Initialization config dict.
+            Noted that, in `DiffuserWrapper`, if you want to load pretrained
+            weight of `repo_id`, please use `no_loading=False`. If you want to
+            modify the `from_pretrained` behavior, you should set `type` as
+            `Pretrained` and set corresponding arguments in `init_cfg`.
+            e.g. (`dict(type='Pretrained', cache_dir='~/.cache/OpenMMLab/')`)
+            Defaults to None.
+
+        *args, **kwargs: Arguments for `module_cls`.
+    """
+
+    def __init__(self,
+                 repo_id: Optional[Union[str, os.PathLike]] = None,
+                 no_loading: bool = False,
+                 init_cfg: Union[dict, List[dict], None] = None,
+                 *args,
+                 **kwargs):
+        super().__init__(init_cfg)
+
+        self._repo_id = repo_id
```

Comment:
I think it's better to follow diffusers convetion to use `from_pretrained` instead of `repo_id`

Reply:
we can refer to https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/modeling_utils.py#L320